### ISSUE TITLE
Provide images with rustup / cargo installed

### ISSUE THREAD

**Issue by Emilgardis** [cross-rs/cross#1516]
We can, before https://github.com/cross-rs/cross/blob/19be83481fd3e50ea103d800d72e0f8eddb1c90c/.github/workflows/ci.yml#L242
also tag and publish an image which has rustup installed, this would help the use case of just wanting an image with cross-compilation tools + rust installed.

related to #1187

**Comment by randomairborne** [comment]
A significant problem with this might be the creation of an excessive number of new dockerfiles. Then again, maybe it could be done programatically somehow.

**Comment by randomairborne** [comment]
How does:
```dockerfile
FROM ghcr.io/cross-rs/{{ARCH}}-unknown-linux-musl:edge

ENV PATH="/root/.cargo/bin:$PATH"

RUN curl -fsSL https://sh.rustup.rs | bash -s -- -y --profile minimal
RUN rustup target add {{ARCH}}-unknown-linux-musl
```
look?

**Comment by randomairborne** [comment]
Actually, that won't work. Some platforms have target triples, others have target quadruples.

**Comment by Emilgardis** [comment]
No, I don't see why we'd need to generate a bunch of dockerfiles. Instead we could just use a build-arg approach.

```Dockerfile
ARG CROSS_IMAGE
ARG TARGET
FROM CROSS_IMAGE
ENV PATH="/root/.cargo/bin:$PATH"
RUN curl -fsSL https://sh.rustup.rs/ | bash -s -- -y --profile minimal
RUN rustup target add $TARGET
```

and then invoke it like `docker build buildx -f Dockerfile.cross-with-cargo --build-arg CROSS_IMAGE="<image>" --build-arg TARGET="<target>"`

hcl would probably be the best solution coupled with https://github.com/cross-rs/cross/pull/849 however

**Comment by randomairborne** [comment]
That was what my comment originally was, and is what I'm currently using for cross-cargo. I just laser-focused on that for some reason LOL!
Is there already a list of supported targets that could be read from to generate the ARGs?

**Comment by Emilgardis** [comment]
All targets built with ci lives in https://github.com/cross-rs/cross/blob/main/targets.toml, and if integrated in this repos CI the target name would already exist in `${{ matrix.target }}`

**Comment by Emilgardis** [comment]
or here also: https://github.com/cross-rs/cross/blob/main/src/docker/provided_images.rs
